### PR TITLE
[Profiler] Add `disabled` and `global` methods to ProfilerConfig.

### PR DESCRIPTION
**Pull request by robieta**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83965
* #83964
* #83895
* #83894
* #83893
* #83892
* __->__ #83891
* #83890
* #83963
* #83622

`ProfilerState::Disabled` and `ProfilerState::KINETO_ONDEMAND` have special semantics. The former is somewhat intuitive, but the degree of behavior branching on the latter (and why the branching is necessary) is less clear. By factoring the enum checks into methods, we can both clairify intent and future proof in case we ever add other global profiling contexts.

Differential Revision: [D38917980](https://our.internmc.facebook.com/intern/diff/D38917980/)

@diff-train-skip-merge

### CODE REVIEW

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 1]
```diff
@@ -558,13 +558,18 @@ void enableProfiler(
...
       config.state == ProfilerState::KINETO ||
-      config.state == ProfilerState::KINETO_GPU_FALLBACK ||
-      config.state == ProfilerState::KINETO_ONDEMAND);
+      config.state == ProfilerState::KINETO_GPU_FALLBACK || config.global());
   TORCH_CHECK(
       !activities.empty(), "No activities specified for Kineto profiler");
 
-  if (config.state == ProfilerState::KINETO ||
-      config.state == ProfilerState::KINETO_GPU_FALLBACK) {
+  if (config.global()) {
```
Comment (slgong-fb) [inline torch/csrc/autograd/profiler_kineto.cpp:565]: Should this line be like below? `ProfilerState::KINETO` seems to be missed.
```
if (config.state == ProfilerState::KINETO || config.global()) {
```

**Review by slgong-fb — APPROVED** [review]
LGTM except for one 'if' statement (comment added).

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 1 above)
Comment (robieta) [reply torch/csrc/autograd/profiler_kineto.cpp:565]: No. `config.global()` is on-demand. `ProfilerState::KINETO` and `ProfilerState::KINETO_GPU_FALLBACK` are covered by the else branch.

**Pull request closed without merging**